agenthub/codeact_agent/codeact_agent.py

```diff
@@ -225,25 +253,10 @@ def _get_messages(self, state: State) -> list[Message]:
             ),
             None,
         )
-
-        # Get the last user text inside content
         if latest_user_message:
-            latest_user_message_text = next(
-                (
-                    t
-                    for t in reversed(latest_user_message.content)
-                    if isinstance(t, TextContent)
-                )
-            )
-            # add a reminder to the prompt
             reminder_text = f'\n\nENVIRONMENT REMINDER: You have {state.max_iterations - state.iteration} turns left to complete the task. When finished reply with <finish></finish>.'
-
-            if latest_user_message_text:
-                latest_user_message_text.text = (
-                    latest_user_message_text.text + reminder_text
-                )
-            else:
-                latest_user_message_text = TextContent(text=reminder_text)
-                latest_user_message.content.append(latest_user_message_text)
+            latest_user_message.content.append(
+                Message(role='user', content=[TextContent(text=reminder_text)])
+            )
```

Comment:
```suggestion
            latest_user_message.content.append(TextContent(text=reminder_text))
```
Let's add a TextContent to the list of TextContexts, or alternatively, the text can be added to the actual text.